conf/local.conf.sample

```diff
@@ -296,14 +293,16 @@ BB_DISKMON_DIRS = "\
 # Use this to add extra packages to the split-debug filesystem
 #EXTRA_DEBUG_PACKAGES = "ltrace strace"
 
-# Uncomment following two lines to enable INITRAMFS. This adds dmcrypt-initramfs-image in 
-# INITRAMFS_IMAGE if encrypted-fs image feature is enabled otherwise add core-image-minimal-initramfs.
-#INITRAMFS_IMAGE = "${@bb.utils.contains('EXTRA_IMAGE_FEATURES', 'encrypted-fs', 'dmcrypt-initramfs-image', 'core-image-minimal-initramfs', d)}"
+# Uncomment to enable encrypted-fs Machine Feature:
+#MACHINE_FEATURES_append = " encrypted-fs"
+# Uncomment it to enable INITRAMFS:
+#INITRAMFS_IMAGE = "${@bb.utils.contains('MACHINE_FEATURES', 'encrypted-fs', 'dmcrypt-initramfs-image', 'core-image-minimal-initramfs', d)}"
+# Uncomment it to bundle initramfs image with kernel image:
 #INITRAMFS_IMAGE_BUNDLE = "1"
 # removing 96boards-tools package so that rootfs does not occupy entire available space over the boot media.
-MACHINE_EXTRA_RRECOMMENDS_remove = "${@bb.utils.contains('EXTRA_IMAGE_FEATURES', 'encrypted-fs', '96boards-tools', '', d)}"
+MACHINE_EXTRA_RRECOMMENDS_remove = "${@bb.utils.contains('MACHINE_FEATURES', 'encrypted-fs', '96boards-tools', '', d)}"
 # Add initramfs image to the boot partition.
-IMAGE_BOOT_FILES_append = "${@bb.utils.contains('EXTRA_IMAGE_FEATURES', 'encrypted-fs', '${dm-initramfs}', '', d)}"
+IMAGE_BOOT_FILES_append = " ${@bb.utils.contains('MACHINE_FEATURES', 'encrypted-fs', '${dm-initramfs}', '', d)}"
```

Comment:
Again, how about doing this through mel.conf? Also please define
`dm-initramfs ?= ""`
somewhere in the configuration so the build wouldn't break for BSPs not defining the variable as investigated after a lot of effort :smile:

Reply:
Yes @abelal, I'd have a discussion with @Noor-Ahsan hopefully today regarding moving these lines to mel.conf.
Secondly, I had a discussion with @ArsalanHAwan. We may drop dm-initramfs in the improvements to come in a few days. I am looking into a composite of yocto's variables that can do the same job as dm-initramfs is doing. Composite will be like this: 
${KERNEL_IMAGETYPE}-${INITRAMFS_NAME}
But I've to take care of some factors, as this change may affect the base release bsps (uno, zcu102, ultrazed) whose documentation is already done according to the flow brought in by dm-initramfs.